homeassistant/components/homekit/util.py

```diff
@@ -90,6 +92,7 @@
         vol.Optional(CONF_MAX_FPS, default=DEFAULT_MAX_FPS): cv.positive_int,
         vol.Optional(CONF_AUDIO_MAP, default=DEFAULT_AUDIO_MAP): cv.string,
         vol.Optional(CONF_VIDEO_MAP, default=DEFAULT_VIDEO_MAP): cv.string,
+        vol.Optional(CONF_VIDEO_CODEC, default=DEFAULT_VIDEO_CODEC): cv.string,
```

Comment:
Since `homekit` is expecting a specific video codec the only option that would work is `copy` so maybe this should be a `CONF_VIDEO_COPY` and make it a `bool` ?

Reply:
I just noticed some of my cams can send opus audio natively as well - maybe add a similar `CONF_AUDIO_COPY` bool config?

Reply:
@xdissent @stickpin Camera support is merged upstream so you should be able to send a PR now.

Reply:
Was able to find a camera that worked with this.  152% -> 3% cpu usage.

Reply:
> Since `homekit` is expecting a specific video codec the only option that would work is `copy` so maybe this should be a `CONF_VIDEO_COPY` and make it a `bool` ?

I think it better to keep an option to set a codec, because for example on RPi and for sure depends on your build of ffmpeg you can choose between two codecs: 
- libx264 - No HW acceleration
- h264_omx - With HW acceleration

Reply:
> I just noticed some of my cams can send opus audio natively as well - maybe add a similar `CONF_AUDIO_COPY` bool config?

I will add it, no problem.

Reply:
> Was able to find a camera that worked with this. 152% -> 3% cpu usage.

Yes, that correct, especially relevant when your caremas are running on some low-end hardware like RPi.

Reply:
@xdissent, @bdraco just submitted the [PR](https://github.com/home-assistant/core/pull/35238).